components/datetime/benches/datetime.rs

```diff
@@ -0,0 +1,121 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::fmt::Write;
+
+use icu_datetime::date::DateTime;
+use icu_datetime::options::{DateStyle, DateTimeFormatOptions, TimeStyle};
+use icu_datetime::provider::DummyDataProvider;
+use icu_datetime::DateTimeFormat;
+
+fn datetime_benches(c: &mut Criterion) {
+    let datetimes = vec![
+        DateTime::new(2001, 9, 8, 18, 46, 40),
+        DateTime::new(2017, 7, 13, 19, 40, 0),
+        DateTime::new(2020, 9, 13, 5, 26, 40),
+        DateTime::new(2021, 1, 6, 22, 13, 20),
+        DateTime::new(2021, 5, 2, 17, 0, 0),
+        DateTime::new(2021, 8, 26, 10, 46, 40),
+        DateTime::new(2021, 12, 20, 3, 33, 20),
+        DateTime::new(2022, 4, 14, 22, 20, 0),
+        DateTime::new(2022, 8, 8, 16, 6, 40),
+        DateTime::new(2033, 5, 17, 20, 33, 20),
+    ];
+    let values = &[
+        ("pl", DateStyle::Full, TimeStyle::None),
+        ("pl", DateStyle::Long, TimeStyle::None),
+        ("pl", DateStyle::Medium, TimeStyle::None),
+        ("pl", DateStyle::Short, TimeStyle::None),
+        ("pl", DateStyle::None, TimeStyle::Full),
+        ("pl", DateStyle::None, TimeStyle::Long),
+        ("pl", DateStyle::None, TimeStyle::Medium),
+        ("pl", DateStyle::None, TimeStyle::Short),
+        ("pl", DateStyle::Full, TimeStyle::Full),
+        ("pl", DateStyle::Long, TimeStyle::Long),
+        ("pl", DateStyle::Medium, TimeStyle::Medium),
+        ("pl", DateStyle::Short, TimeStyle::Short),
+    ];
+
+    let mut results = vec![];
+
+    for _ in 0..datetimes.len() {
+        results.push(String::new());
+    }
+
+    let dp = DummyDataProvider::default();
+
+    {
+        let mut group = c.benchmark_group("datetime");
+
+        group.bench_function("DateTimeFormat/format_to_write", |b| {
+            b.iter(|| {
+                for value in values {
+                    let options = DateTimeFormatOptions {
+                        date_style: value.1,
+                        time_style: value.2,
+                    };
+                    let dtf = DateTimeFormat::try_new(&dp, &options);
```

Comment:
Nit: we will also need skeleton support at some point.

Reply:
Added!